stairway/src/main/java/bio/terra/stairway/impl/FlightDao.java

```diff
@@ -640,6 +640,40 @@ private FlightContextImpl resumeInner(String stairwayId, String flightId)
     }
   }
 
+  void storePersistedStateMap(String flightId, FlightMap persistedStateMap)
+      throws StairwayException, DatabaseOperationException, InterruptedException {
+    DbRetry.retryVoid(
+        "flight.storePersistedStateMap",
+        () -> storePersistedStateMapInner(flightId, persistedStateMap));
+  }
+
+  private void storePersistedStateMapInner(String flightId, FlightMap persistedStateMap)
+      throws SQLException {
+    final String sqlUpsert =
+        "INSERT INTO "
+            + FLIGHT_PERSISTED_TABLE
+            + "(flightId, key, value) VALUES (:flightId, :key, :value1) "
+            + "ON CONFLICT ON CONSTRAINT pk_flightpersisted "
+            + "DO UPDATE SET value = :value2";
+
+    List<FlightInput> inputList = FlightMapUtils.makeFlightInputList(persistedStateMap);
+
+    try (Connection connection = dataSource.getConnection();
+        NamedParameterPreparedStatement statement =
+            new NamedParameterPreparedStatement(connection, sqlUpsert)) {
+      startTransaction(connection);
+      statement.setString("flightId", flightId);
+
+      for (FlightInput input : inputList) {
+        statement.setString("key", input.getKey());
+        statement.setString("value1", input.getValue());
```

Comment:
why are both value1 and value2 set to the same value?

Reply:
I added a comment to explain. `NamedParameterPreparedStatement` is not the Spring class, but one private to Stairway that is not as smart.